components/dashboard/TextArea/PostButtons.tsx

```diff
@@ -96,6 +96,10 @@ const PostButtons = ({ shiftRight, editor }: PostButtonsProps) => {
     setSynced(true);
   };
 
+  useEffect(() => {
+    saveNoteHandler();
+  }, [postPublic]);
+
```

Comment:
We'd actually prefer this to be not in a useEffect. We shouldn't be triggering a function unnecessarily on every change of the state when we can make sure that it triggers right when we need it.